spark/src/main/java/io/ddf/spark/content/SchemaHandler.java

```diff
@@ -1,386 +1,466 @@
+    Tuple2<List<Tuple2<Integer, Column>>, StructType> convertInfo = getConvertInfo(this.getSchema(), schema);
+    List<Tuple2<Integer, Column>> listConvertColumns = convertInfo._1();
+    StructType newSchema =convertInfo._2();
+    try {
+      SparkDDFManager manager = (SparkDDFManager) this.getManager();
+      Accumulator<ApplySchemaStatistic> statisticAccumulator = manager.getSparkContext().accumulator(new ApplySchemaStatistic(), ApplySchemaStatisticParam.getInstance());
+
+      JavaRDD<Row> rdd = ((SparkDDF) this.getDDF()).getRDD(Row.class).toJavaRDD();
+
+
+//      JavaRDD<Row> appliedRdd = rdd.map(row -> convertRowType(row, listConvertColumns, statisticAccumulator, newSchema,isDropRow));
+      JavaRDD<Row> appliedRdd =rdd.map(row -> {
+        ApplySchemaUtils utils = new ApplySchemaUtils();
+        return utils.convertRowType(row, listConvertColumns, statisticAccumulator, newSchema,isDropRow);
+      });
+      if (isDropRow) appliedRdd = appliedRdd.filter(row -> {
+        return row != null;
+      });
+      long totalLineSuccess = appliedRdd.count();
+      ApplySchemaStatistic statistic = statisticAccumulator.value();
+      statistic.setTotalLineSuccessed(totalLineSuccess);
+      if (mLog.isDebugEnabled()) {
+        mLog.debug("ApplySchema in " + (System.currentTimeMillis() - startApplySchema) + " ms");
+        mLog.debug("Total Line Processed: " + statistic.getTotalLineProcessed());
+        mLog.debug("Total Line Succeed: " + statistic.getTotalLineSuccessed());
+        if (mLog.isTraceEnabled()) {
+          statistic.getMapColumnStatistic().forEach((name, columnStatistic) -> {
+            mLog.trace("column: " + name + " fail in: " + columnStatistic.getNumConvertedFailed());
+            mLog.trace("sample: " + org.apache.commons.lang.StringUtils.join(columnStatistic.getSampleData(), ','));
+          });
+        }
+      }
+      if (statistic.getTotalLineSuccessed() > 0) {
+        DataFrame dataFrame = manager.getHiveContext().createDataFrame(appliedRdd.rdd(), appliedRdd.first().schema());
+        return new Tuple2<>((SparkDDF) SparkUtils.df2ddf(dataFrame, manager), statistic);
+      } else {
+        return new Tuple2<>(null, statistic);
+      }
+    } catch (ClassCastException cce) {
+      mLog.error("Exception when down cast from DDFManager to SparkDDFManager", cce);
+      throw new DDFException("applySchema only works with SparkDDFManager");
+    }
+
+  }
+
+  static class ApplySchemaUtils {
+    /**
+     * Convert Row Data To Given Type
+     *
+     * @param row                  which hold multiple columns to convert
+     * @param listConvertColumns   given columnIndex & columnName &  type to convert to
+     * @param statisticAccumulator statistic accumulator
+     */
+    private Row convertRowType(Row row, List<Tuple2<Integer, Column>> listConvertColumns,
+                               Accumulator<ApplySchemaStatistic> statisticAccumulator, StructType newSchema, boolean isDropRow) {

```

Comment:
we already have different methods for dropping rows with NA values in DropNAHandler, don't need to overloading the convertRowType function.